lib/kafka/cluster.rb

```diff
@@ -66,6 +66,36 @@ def get_leader(topic, partition)
       connect_to_broker(get_leader_id(topic, partition))
     end
 
+    def get_group_coordinator(group_id:)
+      @logger.debug "Getting group coordinator for `#{group_id}`"
+
+      refresh_metadata_if_necessary!
+
+      cluster_info.brokers.each do |broker_info|
+        begin
+          broker = connect_to_broker(broker_info.node_id)
+          response = broker.find_group_coordinator(group_id: group_id)
+
+          Protocol.handle_error(response.error_code)
+
+          coordinator_id = response.coordinator_id
+          coordinator = connect_to_broker(coordinator_id)
+
+          @logger.debug "Coordinator for group `#{group_id}` is #{coordinator}"
+
+          return coordinator
+        rescue GroupCoordinatorNotAvailable
+          @logger.debug "Coordinator not available; retrying in 1s"
+          sleep 1
+          retry
+        rescue => e
```

Comment:
feels like catching nearly all kinds of exceptions here is a thing for improvement in later PRs at the very least